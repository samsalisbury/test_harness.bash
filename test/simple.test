#!/usr/bin/env bash

# These are the fundamental tests. This test file itself is tested specially, ensuring
# that all 3 tests are run and that we exit with a zero exit code, both when running this
# by itself, and when running with ./test_harness.bash.

set -euo pipefail

# shellcheck disable=SC1090
source "${TESTHARNESS:-../test_harness.bash}"

(
  begin_test ok/empty
)

(
  begin_test ok/true
  true
)

(
  begin_test ok/run-echo
  run echo 1
)
