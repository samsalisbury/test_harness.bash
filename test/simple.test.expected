=== RUN   simple/ok/empty
--- PASS: simple/ok/empty
=== RUN   simple/ok/true
--- PASS: simple/ok/true
=== RUN   simple/ok/run-echo
--- PASS: simple/ok/run-echo
    $ echo 1
    1
=== RUN   simple/TestGoLikeTest_Pass
--- PASS: simple/TestGoLikeTest_Pass
    $ echo OK
    OK
PASS
ok        simple
